### PR TITLE
abstracted recipe creation and model specification 

### PR DESCRIPTION
**Pull request by a-kong**
Please review the test coverage to make sure I didn't miss anything!

### CODE REVIEW

File: src/R/knn_spec.R [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#' A model specification for k-nearest neighbours
+#'
+#' Create a k-nearest neighbour regression model specification, with the given
+#' neighbours, using rectangular kernel function to weight distances between
+#' samples.
+#'
+#' @param neighbours A positive integer value
```
Comment (samzzzzzh) [inline src/R/knn_spec.R:7]: So a natural number?

File: src/R/knn_spec.R [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+#'
+#' @return A k-nearest neighbour model specification, stating the arguments used:
+#' neighbours and weight_func and the computational engine.
+#'
+#' @export
+#'
+#' @examples
+#' knn_spec(5)
+knn_spec <- function(neighbours) {
```
Comment (Jaskaran1116) [inline src/R/knn_spec.R:16]:
```suggestion
knn_spec <- function (neighbours) {
```

**Review by samzzzzzh — APPROVED** [review]
Everything looks good to me. Thanks for the hard work Amy!

File: src/R/recipe_scale_center.R [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    stop("`recipe_scale_center` expects a data frame")
+  }
+
+  if (class(formula) != "formula") {
+    stop("`recipe_scale_center` expects a formula in the form
+         `target ~ predictor(s)")
+  }
+  recipes::recipe(formula, data = dataframe) %>%
+    recipes::step_center(recipes::all_predictors()) %>%
+    recipes::step_scale(recipes::all_predictors())
```
Comment (Jaskaran1116) [inline src/R/recipe_scale_center.R:29]:
```suggestion
    
    recipes::recipe(formula, data = dataframe) %>%
    recipes::step_center(recipes::all_predictors()) %>%
    recipes::step_scale(recipes::all_predictors())
```

**Review by Jaskaran1116 — APPROVED** [review]

File: src/R/knn_spec.R (hunk 1 above)
Comment (a-kong) [reply src/R/knn_spec.R:7]: yeah hahaha

**Pull request merged**